enterprise/server/raft/rangelease/rangelease.go

```diff
@@ -52,11 +52,15 @@ type Lease struct {
 	leaseRecord *rfpb.RangeLeaseRecord
 
 	timeUntilLeaseRenewal time.Duration
-	stopped               bool
-	quitLease             chan struct{}
+
+	stopMutex sync.Mutex
```

Comment:
Carving out a separate mutex for the stopped field looks good to me. Does that fix the race in your tests?

Since you always cancel/renew the context when you stop/extend the lease, could you replace `stopped` with a check for whether the context is done?

Reply:
I didn't see the timeout this morning when I re-run TestCleanupExpiredSessions:
https://app.buildbuddy.io/invocation/aa6f25d1-4bfc-4a3e-a325-a2203cf623a5 and 
https://app.buildbuddy.io/invocation/2b95abbc-ea60-471a-b5e1-4ba92f72a8ca